packages/perseus/src/widgets/interactive-graphs/graphs/segment.tsx

```diff
@@ -0,0 +1,105 @@
+import {Line} from "mafs";
+import * as React from "react";
+
+import {normalizeCoords, normalizePoints, useInteractivePoint} from "../utils";
+
+import type {Coord} from "../../../interactive2/types";
+import type {PerseusGraphTypeSegment} from "../../../perseus-types";
+import type {MafsGraphProps} from "../types";
+
+export type SegmentProps = MafsGraphProps<PerseusGraphTypeSegment>;
+
+const getSegmentCoords = (
+    props: SegmentProps,
+): ReadonlyArray<ReadonlyArray<Coord>> => {
+    const coords = props.graph.coords;
+    if (coords) {
+        return coords;
+    }
+
+    const ys = (n?: number) => {
+        switch (n) {
+            case 2:
+                return [5, -5];
+            case 3:
+                return [5, 0, -5];
+            case 4:
+                return [6, 2, -2, -6];
+            case 5:
+                return [6, 3, 0, -3, -6];
+            case 6:
+                return [5, 3, 1, -1, -3, -5];
+            default:
+                return [5];
+        }
+    };
+
+    return ys(props.graph.numSegments).map((y) => {
+        let segment: Coord[] = [
+            [-5, y],
+            [5, y],
+        ];
+        segment = normalizeCoords(segment, props.range);
+        segment = normalizePoints(props.range, props.step, segment);
+        return segment;
+    });
+};
+
+export const SegmentGraph = (props: SegmentProps) => {
+    const segments = getSegmentCoords(props);
+
+    React.useEffect(() => {
+        // set initial state
+        props.onGraphChange((current) => ({
+            ...current,
+            coords: segments,
+        }));
+    });
```

Comment:
Should `segments` be a useEffect dep?

Reply:
I'm reworking this area now that I see there's a requirement that a graph should not be graded before it's been interacted with.